internal/installer/installer.go

```diff
@@ -1,5 +1,56 @@
 package installer
 
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+)
+
 var Deps = []string{
 	"cmd", "internal", "pkg", "util", "third-party/minijail",
 }
+
+// GetInstallDir returns the absolute path to the installation
+// directory which is chosen according if $XDG_DATA_HOME is set
+// or the user is running the command as root.
+func GetInstallDir() (string, error) {
+	var installDir string
+
+	xdgPath, xdgSet := os.LookupEnv("XDG_DATA_HOME")
+
+	switch {
+	case xdgSet:
+		installDir = filepath.Join(xdgPath, "cifuzz")
+	case os.Getuid() == 0:
+		installDir = string(os.PathSeparator) + filepath.Join("opt", "code-intelligence", "cifuzz")
```

Comment:
Lets switch these two cases: If we're run as root, we should always do a system-wide installation and therefore not use `$XDG_DATA_HOME`, which is for user-specific files.

```suggestion
	case os.Getuid() == 0:
		installDir = string(os.PathSeparator) + filepath.Join("opt", "code-intelligence", "cifuzz")
	case xdgSet:
		installDir = filepath.Join(xdgPath, "cifuzz")
```